### PR TITLE
`1ES-Template` Conversion for `C`

### PR DESCRIPTION
**Pull request by scbedd**

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 1]
```diff
@@ -1,30 +1,53 @@
...
         Linux_ubuntu_2204_x64_with_iot_samples:
-          Pool: azsdk-pool-mms-ubuntu-2204-general
-          OSVmImage: ubuntu-22.04
+          Pool: azsdk-pool-mms-ubuntu-2204-1es-pt
+          OSVmImage: azsdk-pool-mms-ubuntu-2204-1espt
           VCPKG_DEFAULT_TRIPLET: 'x64-linux'
           build.args: ' -DTRANSPORT_PAHO=ON'
           test_type: 'iot'
           os: 'linux'
+        # Disable live testing for these matrix entries until we have something to
```
Comment (scbedd) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml:24]: Just keeping the commented builds around [from main](https://github.com/Azure/azure-sdk-for-c/blob/2dce952c305249fe24f848d09e5acb957240af9e/eng/pipelines/templates/jobs/archetype-sdk-tests.yml#L51)

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 2]
```diff
@@ -1,30 +1,53 @@
...
-  - job: LiveTest
+  - job:
+    displayName: LiveTest
     condition: and(succeededOrFailed(), ne(variables['Skip.Test'], 'true'))
     strategy:
       matrix:
         Linux_x64_with_iot_samples:
-          Pool: azsdk-pool-mms-ubuntu-2004-general
-          OSVmImage: ubuntu-20.04
+          Pool: $(LINUXPOOL)
```
Comment (scbedd) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml:11]: Before someone asks about needing to use the variable...I think that variable replacement only happens on the very surface level of the template construct. 

[Commit](https://github.com/Azure/azure-sdk-for-c/commit/a787af99bb9e65938c7c5b91b7b6cd69954bbc65)
[Build](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3616204&view=results)

Notice how `$(LINUXVMIMAGE)` isn't being honored? It simply isn't replacing it when the pool is one level deeper than the standard.

Comment (scbedd) [reply eng/pipelines/templates/jobs/archetype-sdk-tests.yml:11]: No idea why, reverting the changes though.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -183,7 +183,7 @@ When code samples take dependencies, readers should be able to use the material
...
+- **Open-source** : Open source offerings that use an [Open Source Initiative (OSI) approved license](https://opensource.org/licenses). Any component whose license isn't OSI-approved is considered a commercial offering. Prefer OSS projects that are members of any of the [OSS foundations that Microsoft is part of](https://opensource.microsoft.com/ecosystem/). Prefer permissive licenses for libraries, like [MIT](https://opensource.org/licenses/MIT) and [Apache 2](https://opensource.org/licenses/Apache-2.0). Copy-left licenses like [GPL](hhttps://opensource.org/license/gpl-3-0) are acceptable for tools, and OSs. [Kubernetes](https://github.com/kubernetes/kubernetes), [Linux](https://github.com/torvalds/linux), and [Newtonsoft.Json](https://github.com/JamesNK/Newtonsoft.Json) are examples of this license type. Links to open source components should be to where the source is hosted, including any applicable license, such as a GitHub repository (or similar).
```
Comment (benbp) [inline CONTRIBUTING.md:186]:
```suggestion
- **Open-source** : Open source offerings that use an [Open Source Initiative (OSI) approved license](https://opensource.org/licenses). Any component whose license isn't OSI-approved is considered a commercial offering. Prefer OSS projects that are members of any of the [OSS foundations that Microsoft is part of](https://opensource.microsoft.com/ecosystem/). Prefer permissive licenses for libraries, like [MIT](https://opensource.org/licenses/MIT) and [Apache 2](https://opensource.org/licenses/Apache-2.0). Copy-left licenses like [GPL](https://opensource.org/license/gpl-3-0) are acceptable for tools, and OSs. [Kubernetes](https://github.com/kubernetes/kubernetes), [Linux](https://github.com/torvalds/linux), and [Newtonsoft.Json](https://github.com/JamesNK/Newtonsoft.Json) are examples of this license type. Links to open source components should be to where the source is hosted, including any applicable license, such as a GitHub repository (or similar).
```

**Review by benbp — APPROVED** [review]

**Pull request merged**